src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        Date fromDate;
+        Date toDate;
+        try {
+            fromDate = dateFormat.parse(dateFrom);
+            toDate = dateFormat.parse(dateTo);
+        } catch (ParseException e) {
+            throw new RuntimeException("Incorrect date format", e);
+        }
+
+        Map<String, Integer> salaries = new HashMap<>();
```

Comment:
Instead of using a HashMap to store salaries, consider using a simpler data structure since the use of HashMap was advised against in the checklist.